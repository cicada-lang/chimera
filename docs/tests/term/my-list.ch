clause List(List::Null())
clause List(List::Cons(_head, tail)) -- { List(tail) }

print find _ {
  List(List::Null())
  List(List::Cons(1, List::Cons(2, List::Cons(3, List::Null()))))
}

print find [list, x, y, z] {
  Equal(list, List::Cons(1, List::Cons(2, List::Cons(3, List::Null()))))
  Equal(list, List::Cons(x, List::Cons(y, List::Cons(z, List::Null()))))
}

print find [list, x, y, z] {
  Equal(list, List::Cons(x, List::Cons(y, List::Cons(z, List::Null()))))
}
