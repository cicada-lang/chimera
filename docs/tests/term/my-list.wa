clause List(null())
clause List(cons(_head, tail)) -- { List(tail) }

print find _ {
  List(null())
  List(cons(1, cons(2, cons(3, null()))))
}

print find [list, x, y, z] {
  list = cons(1, cons(2, cons(3, null())))
  list = cons(x, cons(y, cons(z, null())))
}

print find [list, x, y, z] {
  list = cons(x, cons(y, cons(z, null())))
}
