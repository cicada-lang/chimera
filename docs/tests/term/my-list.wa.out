[_.0]
[[cons(1, cons(2, cons(3, null()))), 1, 2, 3]]
[[cons(_.0, cons(_.1, cons(_.2, null()))), _.0, _.1, _.2]]
